modules/javafx.graphics/src/main/java/javafx/css/Match.java

```diff
@@ -42,15 +46,46 @@
  */
 public final class Match implements Comparable<Match> {
 
+    static Match of(Selector selector) {
+        return switch(selector) {
+            case SimpleSelector s -> ofSimpleSelector(s);
+            case CompoundSelector cs -> ofCompoundSelector(cs);
+        };
+    }
+
+    private static Match ofCompoundSelector(CompoundSelector selector) {
+        PseudoClassState pseudoClasses = new PseudoClassState();
+        int idCount = 0;
+        int styleClassCount = 0;
+        List<SimpleSelector> containedSelectors = selector.getSelectors();
+
+        for (int n = 0, max = containedSelectors.size(); n < max; n++) {
+            Match match = ofSimpleSelector(containedSelectors.get(n));
+
+            pseudoClasses.addAll(match.getPseudoClasses());
+            idCount += match.idCount;
+            styleClassCount += match.styleClassCount;
+        }
+
+        return new Match(selector, pseudoClasses, idCount, styleClassCount);
+    }
+
+    private static Match ofSimpleSelector(SimpleSelector selector) {
+        int idCount = selector.getId().isEmpty() ? 0 : 1;
+        int styleClassCount = selector.getStyleClassSet().size();
+
+        return new Match(selector, selector.getPseudoClassStates(), idCount, styleClassCount);
+    }
+
     private final Selector selector;
     private final Set<PseudoClass> pseudoClasses;
 
-    final int styleClassCount;
-    final int idCount;
+    private final int styleClassCount;
+    private final int idCount;
 
     // CSS3 spec gives weight to id count, then style class count,
     // then pseudoclass count, and finally matching types (i.e., java name count)
-    final int specificity;
+    private final int specificity;
 
     @SuppressWarnings("removal")
```

Comment:
What is this removal warning suppression? I know it was there before these changes.

Reply:
`SimpleSelector` was deprecated for removal (from the public javafx.css package). Now that it is moved to the internal package, the deprecation on that was also removed.  The `SuppressWarnings` is therefore no longer needed here as the deprecation is no longer present on the now internal `SimpleSelector` type.

Reply:
The diff seems to show I did not remove that line, but it should no longer be needed.  I'll check.

Reply:
The line was correctly removed.